src/plugins/image_embeddable/public/image_editor/image_editor_flyout.tsx

```diff
@@ -429,6 +431,9 @@ export function ImageEditorFlyout(props: ImageEditorFlyoutProps) {
       {isFilePickerOpen && (
         <FilePicker
           kind={imageEmbeddableFileKind.id}
+          deleteButtonPredicate={(file) =>
```

Comment:
Why does this have to be configured on an embeddable image level?

Should we try to do this on the files level instead to try to avoid making consumer code more complicated with optional dependency on the security plugin? I believe a lot of consumers would duplicate this, but this seems to be a very common use-case.

Maybe we should have something like `files.isMyFIle(file)`. And use it internally inside FilePicker. The prop on FilePicker could be `allowDelete: `none` | `myOnly` | `all` 

Later if we need more flexibility we can add more generic predicate way

WDYT?

Reply:
Yes, I was thinking about some sort of integration like that in the the `FilesClient`, but currently it does not work out well:

- Right now the `FilesClient` actually does not have access to security plugin, hence no access to authenticated user. Currently it only has access to `core.http` and `fileKind: string`.
- The `FilesClient` interface is shared between server and browser: so whatever we build needs to work in both environments.
- There are actually two interfaces: `FilesClient` and `FilesClientBase`, which do similar/the same things. And the one that is used in React components is the "base" one&mdash;it is in a package (not plugin).
  - Then the question is how do we represent authentication/signed in user in package types, so they maybe are not tied too tightly to Kibana security.
  - The user profile type from security is actually in a plugin `@kbn/security-plugin`, which ideally we would need to move to a package.

Reply:
My 2c: deferring this change makes sense to me too. Note: it is now possible to use types from a plugin in a package after the latest round of updates to our TS build process, but I hear what you are saying about it being "more ideal" if those types were in a package.